### PR TITLE
Adds a Join Queue if the server exceeds its hard population cap [Ready]

### PR DESCRIPTION
**Pull request by Ikarrus**
Players can join a queue that will notify them with a message and sound when a slot becomes available. First come first serve.

It even tells you how many other players are ahead of you in the queue.

Players have approximately 20 seconds to join after the notification. Failure to join in that window removes them from the queue, to remove AFKers.

### CODE REVIEW

**Comment by Iamgoofball** [comment]
can I buy VIP access to bypass the queue

when will we have Premium accounts

**Comment by WJohn** [comment]
I have a feeling this will never even be required given that it's almost impossible to have 70 living players in the game at any one time, but I guess it's better to have it.

**Comment by duncathan** [comment]
I had this issue like, last week. It's summer, man.

**Comment by CosmicScientist** [comment]
@WJohn in this thread: https://tgstation13.org/phpBB/viewtopic.php?f=10&t=4120

Scroll down midway to find MSO claiming:
"Currently, there are 3 caps:

Soft cap at 50. This is what I set the banners to. You can join, it just nicely requests you join another server.

Hard cap at 65. This prevents joining the game, but you can observe.

Extreme cap at 70. This prevents connecting to the server. I put this in place after I watched the server buckle at 95 players. while basil had almost none. basil then got 10 new players. It was nice."


I don't know about round start but afterwards you get 65 living players maximum.

**Comment by duncathan** [comment]
Hardcap's at 50 right now, I think.

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -434,3 +438,23 @@ var/datum/subsystem/ticker/ticker
 	if(randomtips.len)
 		world << "<font color='purple'><b>Tip of the round: </b>[html_encode(pick(randomtips))]</font>"
 
+/datum/subsystem/ticker/proc/check_queue()
+	if(!queued_players.len || !config.hard_popcap)
+		return
+
+	queue_delay ++
```
Comment (RemieRichards) [inline code/controllers/subsystem/ticker.dm]: ```queue_delay++``` not ```queue_delay ++```

**Pull request merged**